### DISCUSSION TITLE
GOM error with $$Material_Model

### DISCUSSION THREAD

**Discussion by jacobosaa** [Anura3D/Anura3D_OpenSource#144] [Q&A]
Hello Everyone:

I'm currently studying the flow behavior of thickened tailings, modeling a different set of columns collapsing due to its own weight, comparing the results between modelling the material as a liquid and as a saturated soil. I'm using the 16.0.6 GiD version, with the Anura3D v.2023, using the MPM method. I've just modeled the column using the Mohr-Coulomb constitutive model as a saturated fully coupled soil, and there was no problem during the calculation process. BUT, when it comes to use the Bingham constitutive model, this message appears and i'm not able to continue with the calculation process.
I've already checked if the .dll (A3DBingham.dll) file was in the problemtype folder, which came "naturally" after compiling the code from this forum one week ago. I was using an older version of Anura3D and GiD but then opted for the newest version to get rid of this problem, which happened to me before. 

[PRUEBA2D20.gid.zip](https://github.com/Anura3D/Anura3D_OpenSource/files/13210010/PRUEBA2D20.gid.zip)

![error gom bingham](https://github.com/Anura3D/Anura3D_OpenSource/assets/128545000/81661ca6-7ad1-4354-9aaf-5b216ac93463)

I'll be attaching the problemtype folder, an image of the error message and the model itself. 
Hoping that you can help me, because I really need it.
Thanks.

**Comment by jacobosaa** [comment]
[Anura3D_2023.gid.zip](https://github.com/Anura3D/Anura3D_OpenSource/files/13210071/Anura3D_2023.gid.zip)

**Reply by mikael-onsilva** [reply]
Hi Jacob,

I'm having the same error using a Frictional Fluid.

Have you found out what's causing it?

**Reply by jacobosaa** [reply]
Hi Mikael, thanks for your feedback. 
Unfortunately, I haven't found what is causing this error, BUT, recently I have been told by @emre-ncche that in order to correct the file, we should be able to navigate your .A3D files and search for GOM file. Then change material model from “Bingham” to “bingham_liquid” (which whas my case, maybre in yours you could change it from Frictional Fluid to "frictional_liquid"). It should work.

I tried it and that error didn't appear, but sadly, some other errors appeared. 
Let me know if that helped you.
Best regards from Chile.

Jacobo

**Reply by mikael-onsilva** [reply]
Great, it worked with "frictional_liquid". The results aren't the expected, but it's probably other configurations.

The main question is: why does the program creates the GOM file with the wrong material model name? :)

Best regards.

**Reply by jacobosaa** [reply]
Hi mikael.

Well, I don't know too much about the code within Anura3D. I'd like to know what are you modelling and what kind of unexpected results are you getting.

**Reply by mikael-onsilva** [reply]
I'm gonna try to model a tailings stack susceptible to static liquefaction to run its runout. 

At first I'm just running a column collapse, but changing it from Mohr Coulomb to a Frictional Fluid or Bingham (I'm gonna try both to see wich one suits better), but the material is spreading like water, with no deposition angle. I've just found that the GOM file created by GiD does not creates the lines for the constitutive model of liquids. As I'm not sure how the parameters must be called, I'm starting some trials with:

$$MATERIAL_MODEL_LIQUID
frictional_liquid
$$ELASTIC_YOUNG_MODULUS
10000
$$ELASTIC_POISSON_RATIO
0.5
$$FLUID_FRICTION_ANGLE
30

But I don't know if these are the right variables names. The simulation is running as we speak.

**Reply by jacobosaa** [reply]
Hi mikael! 
I'm currently doing a set of columns made of tailings, with different solid content, from 20% up to 80%, and comparing them between using Mohr Coulomb (saturated fully coupled) and Bingham. Maybe we could help each other. In my case, I'm using different values for Young modulus and Poisson ratio. (E=1500 and Poisson = 0.45)

May I ask you for the "geotechnical" data for your tailings? specifically, for each tailing's solid content, I need the slurry/pulp density, the Bingham yield  stress and viscosity. If you like, we could share the data. 

Hoping that your models runs ok, and to read from you soon.

Jacobo

**Reply by mikael-onsilva** [reply]
Great Jacobo.

Send me an e-mail so we can talk more freely.

mikael.onsilva@gmail.com.

**Comment by gdicarluccio** [comment]
Hi all! Thanks for your feedback! The Problem Type has been just updated.
Could you please check if this fixed your issues with the liquid material?

**Comment by jacobosaa** [comment]
Hello Gaia!
Thanks for your reply, I will check right now. Please, could you tell me if it's needed to compile again that file or I just need to download it and put it in the right folder?

**Reply by gdicarluccio** [reply]
You should open your model in GiD and reload the problem type (Transform to new problemtype). Check that all the conditions are assigned correctly,  create the mesh, save and then generate the files again.
